### PR TITLE
feat(dev/workflow): run gofmt lint on make format

### PR DESCRIPTION
**Pull request by bartsmykla**
It looks like `go fmt ./...` is not enough in regards of formatting our go code to satisfy our linter checks from `make check` target. This PR is adding `gofmt` linter to be run during `make format`, which should sufficiently resolve this problem`

Closes: https://github.com/kumahq/kuma/issues/6124

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues -- https://github.com/kumahq/kuma/issues/6124
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS -- Not apply as it's just an improvement for our dev workflow
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) -- Not apply as it's just an improvement for our dev workflow
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? -- Not apply as it's just an improvement for our dev workflow
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? -- I don't think so
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: mk/check.mk [hunk 1]
```diff
@@ -2,7 +2,7 @@
 fmt: fmt/go fmt/proto ## Dev: Run various format tools
 
 .PHONY: fmt/go
-fmt/go: ## Dev: Run go fmt
+fmt/go: golangci-lint-fmt ## Dev: Run go fmt
 	go fmt ./...
```
Comment (michaelbeaumont) [inline mk/check.mk:6]: is this still needed then?

Comment (bartsmykla) [reply mk/check.mk:6]: Do you mean `go fmt ./...`?

Comment (michaelbeaumont) [reply mk/check.mk:6]: yes, if `golangci-lint` is running `fmt` do we still need it?

Comment (bartsmykla) [reply mk/check.mk]: removed

**Review by michaelbeaumont — COMMENTED** [review]
Hmm I think this means we're running `gofmt` with `golangci-lint` twice when we run `make check`, though, right?

Maybe we should only run `fmt/proto`, not `format`, as a prerequisite of `check` and leave `make format` for devs to run on the CLI?

**Comment by bartsmykla** [comment]
@michaelbeaumont sounds good to me, so I applied your suggestions

**Review by michaelbeaumont — APPROVED** [review]

**Pull request merged**